src/Microsoft.Dnx.Runtime/FrameworkReferenceResolver.cs

```diff
@@ -277,7 +342,7 @@ private static FrameworkInformation GetFrameworkInformation(DirectoryInfo direct
 
                             var entry = new AssemblyEntry();
                             entry.Version = version != null ? Version.Parse(version) : null;
-                            frameworkInfo.Assemblies.Add(assemblyName, entry);
+                            frameworkInfo.Assemblies[assemblyName] = entry;
```

Comment:
We actually do allow that already. We should flow all the information to make sure it's possible. The only change here would be to support multiple versions of assemblies in the framework information assemblies right?